File: lib/java/src/protocol/TList.java
// Copyright (c) 2006- Facebook
// Distributed under the Thrift Software License
//
// See accompanying file LICENSE or visit the Thrift site at:
// http://developers.facebook.com/thrift/

package com.facebook.thrift.protocol;

/**
 * Helper class that encapsulates list metadata.
 *
 * @author Mark Slee <mcslee@facebook.com>
 */
public class TList {
  public TList() {}

  public TList(byte t, int s) {
    elemType = t;
    size = s;
  }

  public byte elemType = TType.STOP;
  public int  size     = 0;
}

File: lib/csharp/src/Protocol/TList.cs
//
//  TList.cs
//
//  Begin:  Aug 19, 2007
//  Authors:
//		Todd Berman <tberman@imeem.com>
//
//  Distributed under the Thrift Software License
//
//  See accompanying file LICENSE or visit the Thrift site at:
//  http://developers.facebook.com/thrift/using

using System;
using System.Collections.Generic;
using System.Text;

namespace Thrift.Protocol
{
	public struct TList
	{
		private TType elementType;
		private int count;

		public TList(TType elementType, int count)
			:this()
		{
			this.elementType = elementType;
			this.count = count;
		}

		public TType ElementType
		{
			get { return elementType; }
			set { elementType = value; }
		}

		public int Count
		{
			get { return count; }
			set { count = value; }
		}
	}
}
